src/components/schedule-visualizer/colors.ts

```diff
@@ -21,29 +8,81 @@ async function _hash(s: string): Promise<ArrayBuffer> {
 
 export default class ColorManager {
   async robotColor(name: string, model: string): Promise<string> {
-    const key = this._robotKey(name, model);
-    let color = this._robotColorCache[key];
+    let color = this._robotColorCache[name];
     if (!color) {
       const modelHash = new Uint16Array(await _hash(model));
-      const hue = modelHash[0] % 360;
       const nameHash = new Uint16Array(await _hash(name));
-      const satlum = nameHash[0] % 2500;
-      const saturation = 50 + (satlum % 50);
-      const luminance = 25 + satlum / 50;
-      color = `hsl(${hue}, ${saturation}%, ${luminance}%)`;
-      this._robotColorCache[key] = color;
+      color = ColorManager._getLightColor(modelHash[0], nameHash[0]);
+      this._robotColorCache[name] = color;
     }
     return color;
   }
 
-  robotColorFromCache(name: string, model: string): string | null {
-    const color = this._robotColorCache[this._robotKey(name, model)];
-    return color ? color : null;
+  async robotTrajectoryColor(name: string, model: string): Promise<string> {
```

Comment:
We could probably combine this with `robotImageColor`

```ts
async robotPrimaryColor(fleet: string, name: string, model: string, image?: string): Promise<string | undefined>
```

What do you think of this?

Reply:
I'm good with this, but maybe without the `fleet` parameter because I dun see it used in either methods.